### ISSUE TITLE
getGroupMembershipForUser returning zero groups for cn's containing quotes

### ISSUE THREAD

**Issue by kmz850** [gheeres/node-activedirectory#118]
It appears that getGroupMembershipForUser is bringing back 0 groups for users that have quotes in their CN ie (Smith, William "Bill").

The line `if (! groups) console.log('User: ' + element + ' not found.');` will not actually fire.

Below is an excerpt of my code. Let me know if I can provide any more details:
```
   async.retry({times: 10, interval: 4000}, function(cbGroupMembership, results) 
        {
          var opts = {
             includeMembership: [ 'group' ]
           }

          ad.getGroupMembershipForUser(opts, element, function(err, groups) 
          {
                if ((err) || (! groups)) {
                  console.log('getGroupMembershipForUser ERROR for ' + element + ': ' + JSON.stringify(err));
                  return cbGroupMembership('error', null); 
                }

                if (! groups) console.log('User: ' + element + ' not found.');
                
                var groupsTemp = new Array();

                _.each(groups, function(groups) {
                  groupsTemp.push(groups.cn);
                });

                var usernameToGroups = {"username": element, "groups": groupsTemp};
                groupMembership.push(usernameToGroups);

                cbGroupMembership(null, 'Success');           
          });
```

**Comment by gheeres** [comment]
Is the "element" value provided by some other external system? Or is it the result of some other AD query? 

Is the "element" a DN, sAMAccountName or userPrincipalName? The simple answer is that the query must be escaped. If you're providing the DN, then it's up to your code to ensure it's properly escaped. However, if you're reusing a DN that was returned via an AD query, then AD will need to handle the escaping.

Please get back to me with more information.

**Comment by kmz850** [comment]
The element is their sAMAccountName and is returned by the ad.find method. Then I'm looping through my employeesArray and passing sAMAccountName into each getGroupMembershipForUser call.

```
ad.find(opts, function(err, results) {
        
               if ((err) || (! results)) {
                  console.log('ERROR for ' + element + ': ' + JSON.stringify(err));
                 
                  return callbackFind('error', null); // don't return cb(err), that will trigger asynch.each final callback
               }
               _.each(results.users, function(user) {
                if (typeof user.sAMAccountName != 'undefined')
                  employeesArray.push(user.sAMAccountName);
               });

               console.log(employeesArray);

               callbackFind(null, 'Success');

});
```

Sample employeesArray:
laveryc
tempemail
keverr
humphred
crowellk

**Comment by kmz850** [comment]
Ok I *think* I resolved the issue with this fix:

```javascript
function parseDistinguishedName(dn) {
  log.trace('parseDistinguishedName(%s)', dn);
  if (! dn) return(dn);

  // Currently a bug in ldapjs that isn't properly escaping DNs with ','
  // in the name.
  dn = dn.replace(/"/g, '\\"');  //escape "
  return(dn.replace('\\,', '\\\\,'));
}
```

**Comment by gheeres** [comment]
Thanks! That looks appropriate. Can you cleanup the code (move / readjust / truncate comments) and submit a pull request?

**Comment by gheeres** [comment]
Fix / merged via pull request #120
